platforms/web/scripts/build-tools/buildTools.ts

```diff
@@ -0,0 +1,122 @@
+import fs from 'fs';
+
+import type { HtmlTagDescriptor } from 'vite';
+import type { Target } from 'vite-plugin-static-copy';
+
+const initSettings = (mode: string) => {
+  const localFile = `ini/.webapp.${mode}.ini`;
+  const templateFile = `ini/templates/.webapp.${mode}.ini`;
+
+  // The build ONLY uses .ini files in /ini to include in the build output.
+  // All .ini files in the directory are git ignored to customer specific values out of source control.
+  // However, this script will automatically create a .ini file for the current mode if it doesn't exist
+  // by copying the corresponding mode file from the ini/templates directory.
+  if (!fs.existsSync(localFile) && fs.existsSync(templateFile)) {
+    fs.copyFileSync(templateFile, localFile);
+  }
+
+  return localFile;
+};
+
+export const getFileCopyTargets = (mode: string): Target[] => {
+  const localFile = initSettings(mode);
+  const fileCopyTargets: Target[] = [
+    {
+      src: localFile,
+      dest: '',
+      rename: '.webapp.ini',
+    },
+  ];
+
+  // These files are only needed in dev / test / demo, so don't include in prod builds
+  if (mode !== 'prod') {
+    fileCopyTargets.push({
+      src: '../../packages/testing/epg/*',
+      dest: 'epg',
+    });
+  }
+
+  return fileCopyTargets;
+};
+
+export const getGoogleVerificationTag = (env: Record<string, string>): HtmlTagDescriptor[] => {
+  if (!env.APP_GOOGLE_SITE_VERIFICATION_ID) return [];
+
+  return [
+    {
+      tag: 'meta',
+      injectTo: 'head',
+      attrs: {
+        content: process.env.APP_GOOGLE_SITE_VERIFICATION_ID,
+        name: 'google-site-verification',
+      },
+    },
+  ];
+};
+
+// @todo: move to common?
+type ExternalFont = {
+  resource: 'google' | 'system';
+  fontFamily: string;
+};
+
+export const extractExternalFonts = (fontEnvVar: string = ''): ExternalFont[] => {
+  if (!fontEnvVar) return [];
+
+  return fontEnvVar.split(',').map((font) => {
+    const [resource, fontFamily] = font.split(':');
+
+    return { resource, fontFamily } as ExternalFont;
+  });
+};
+
+export const getGoogleFontTags = (fonts: ExternalFont[]): HtmlTagDescriptor[] => {
+  if (fonts.length === 0) return [];
+
+  // Currently only supporting google fonts
+  const uniqueFonts = makeFontsUnique(fonts);
```

Comment:
`let uniqueFonts = [...new Set(fonts)];` one line solution?

Reply:
I was willing to implement it, but it doesn't work, because it's not an array of string and we're looking at the `font.fontFamily`.

Reply:
Ah I didnt saw that. It feels a little bit bloated to introduce a specific function which only works with this array, only because a custom property on the object (font.fontFamily) is used.

But since this code is in it's own domain. Let's not waste to much time on this. I think it's acceptable!

Reply:
I do agree though. If you have another suggestion, feel free!

Reply:
But let's merge it now indeed :-)